### PR TITLE
Add typings

### LINKED ISSUE #5: Create TypeScript typings
Create TypeScript typings and add them to the package.json to support more developers.

### PR DESCRIPTION
**Pull request by TheGLander**
Adds typings, fixes #5

### CODE REVIEW

File: multiDict.d.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	DType extends GenericConstructor = typeof Map,
+	Types extends GenericConstructor[] = []
+> {
+	private tree: Tree<Keys, Val, DType, Types>
+	private items: Set<Item<Keys, Val>>
+	/**
+	 * @param entries Iterable of [...keys, value] entries.
+	 * @param options
+	 */
+	constructor(entries: [...Keys, Val][], options?: TreeOptions<DType, Types>)
```
Comment (tomhooijenga) [inline multiDict.d.ts]: `entries` param is not optional but it should be

File: multiDict.d.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	get size(): number
+	get [Symbol.toStringTag](): "MultiDict"
+	set(...keysAndValue: [...Keys, Val]): this
```
Comment (tomhooijenga) [inline multiDict.d.ts]: ` [...Keys, Val]` gives error `TS1256: A rest element must be last in a tuple type.`

File: multiDict.d.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	 */
+	delete(...keys: Keys): boolean
+	/**
+	 * Remove all entries
+	 */
+	clear(): void
+	/**
+	 * Create a new dictionary from the given level.
+	 */
+	level<Newkeys extends any[]>(
```
Comment (tomhooijenga) [inline multiDict.d.ts]: `Newkeys` has a typo.

File: multiDict.d.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+	 * Get an iterator for each of the values.
+	 *
+	 * @generator
+	 */
+	values(): IterableIterator<Val>
+	/**
+	 * Call a callback for each of the registered entries.
+	 *
+	 * @param callback
+	 * @param thisArg Option 'this' context for the callback
```
Comment (tomhooijenga) [inline multiDict.d.ts]: Not your fault, but `Option` should be `Optional`. Could you change this in `src/index.js` as well?

File: multiDict.d.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+	 */
+	forEach<ThisArg>(
+		callback: (this: ThisArg, key: Keys, val: Val, thisArg: this) => void,
```
Comment (tomhooijenga) [inline multiDict.d.ts]: `thisArg` in the callback is not a good name. It is the current dictionary so something like `dict` or `dictionary` would be better.

File: package.json [hunk 6]
```diff
@@ -39,5 +39,6 @@
   "dependencies": {
     "@babel/runtime": "^7.11.2",
     "@teamawesome/access": "^1.0.7"
-  }
+  },
+  "types": "./multiDict.d.ts"
```
Comment (tomhooijenga) [inline package.json]: Could you rename this to  `./multi-dict.d.ts` to keep the name the same as the package?

File: multiDict.d.ts [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+	private root: DType
+	public options: TreeOptions<DType, Types>
+	constructor(options: TreeOptions<DType, Types>)
+
+	get(keys: Keys, value: false): DType | Types[number] | undefined
+	get(keys: Keys, value: true): Val | undefined
+	set(keys: Keys, value: Val): void
+	has(keys: Keys): boolean
+	clear(): void
+	level(keys: Keys): Object[]
```
Comment (tomhooijenga) [inline multiDict.d.ts]: The return type should be `Item[]`

File: multiDict.d.ts [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+	private createNode(depth: number): Object
```
Comment (tomhooijenga) [inline multiDict.d.ts]: I think the return type should be `DType | Types[number]` because `get` also has this.

**Review by tomhooijenga — CHANGES_REQUESTED** [review]
Great work, nearly there!

File: multiDict.d.ts (hunk 2 above)
Comment (zyudneko) [reply multiDict.d.ts]: You must have an old version of typescript installed (in your IDE, prehaps), the latest 4.0 version allows for non-last tuple rests

Comment (tomhooijenga) [reply multiDict.d.ts]: Good catch, my ide is at 3.9.5

File: multiDict.d.ts (hunk 3 above)
Comment (zyudneko) [reply multiDict.d.ts]: I don't see it, what's the typo exactly?

Comment (tomhooijenga) [reply multiDict.d.ts]: Its two words, so it should be `NewKeys` (capital K)

File: package.json [hunk 9]
```diff
@@ -1,43 +1,44 @@
 {
```
Comment (tomhooijenga) [inline package.json]: woah, don't think this was supposed to happen

File: multiDict.d.ts (hunk 3 above)
Comment (zyudneko) [reply multiDict.d.ts]: Ah, yeah

**Review by tomhooijenga — APPROVED** [review]

**Pull request merged**